### PR TITLE
Configuring part in README

### PR DESCRIPTION
**Pull request by mpwp**
Add the overall structure of yi.
It's missing 3 simple useful examples (for BufferM, EditorM and YiM) to be minimally complete.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,18 +15,48 @@ Just doing `stack install yi` would be akin to unwrapping a box of legos and
...
+    
+ * *Actions*, how to interact with the editor and it's the part that makes most of Yi. It is structured around a stack of three monadic DSLs:
+   * **BufferM** for all buffer-local operations, like insertion, deletion of text, and annotation of buffer contents. It can be understood as a monad that encapsulates the state of one buffer.
+   * **EditorM** for editor-level operations, e.g., opening and closing windows and buffers. Operations involving *more than one buffer are handled at this level too.*
+   * **YiM** for IO-level operations. There, one can operate on files, processes, etc. *This is the only level where IO can be performed.*
+
+ * *Keymaps*, how to trigger actions, they serve as the basis to use or to create the ones you need;
+    * [vim-like](https://hackage.haskell.org/package/yi-keymap-vim)
+    * [emacs-like](https://hackage.haskell.org/package/yi-keymap-emacs)
+    * [cua-like](https://hackage.haskell.org/package/yi-keymap-vim)
```
Comment (noughtmare) [inline README.md]: I think this should link to http://hackage.haskell.org/package/yi-keymap-cua.

File: README.md [hunk 2]
```diff
@@ -15,18 +15,48 @@ Just doing `stack install yi` would be akin to unwrapping a box of legos and
...
-### Advanced configuration examples
+## Configuring
+
+A good starting point is choosing an example configuration of your liking in the
+[example-configs][userconfigs] directory, building it, running it, and tinkering with it.
 
-Some people share their yi configuration here on [github](https://github.com/search?utf8=%E2%9C%93&q=yi-config+language%3Ahaskell&type=).
+Yi, as a library, can be categorized into four parts :
+
+* *UIs*, how the editor is rendered following input events;
```
Comment (ethercrow) [inline README.md]: I would say "*Frontends* are responsible for rendering the editor state and passing user events to the core".

File: README.md [hunk 3]
```diff
@@ -15,18 +15,48 @@ Just doing `stack install yi` would be akin to unwrapping a box of legos and
...
+   * **YiM** for IO-level operations. There, one can operate on files, processes, etc. *This is the only level where IO can be performed.*
+
+ * *Keymaps*, how to trigger actions, they serve as the basis to use or to create the ones you need;
+    * [vim-like](https://hackage.haskell.org/package/yi-keymap-vim)
+    * [emacs-like](https://hackage.haskell.org/package/yi-keymap-emacs)
+    * [cua-like](https://hackage.haskell.org/package/yi-keymap-vim)
+  
+* *Glue code*, how the three other parts are shaped together.
+
+The main configuratong work will takes place in the *Actions* and *Keymaps* parts.
```
Comment (ethercrow) [inline README.md]: > configuratong
typo

And well, that really depends. One might go nuts with configuring a keymap, while the other makes new modes, while yet another is making a new frontend. In general, you spend configuration effort in the area that is interesting to you.

File: README.md [hunk 4]
```diff
@@ -15,18 +15,48 @@ Just doing `stack install yi` would be akin to unwrapping a box of legos and
...
+
+So to unleash the power of haskell in text editing, it is possible to define:
```
Comment (ethercrow) [inline README.md]: I think this whole paragraph confuses more than helps.
"2. compositions of buffers (**EditorM**)"
let's take a look at `printMsg :: String -> EditorM ()` for example. In what way is it a "composition of buffers"? I don't understand. I think the breakdown you gave above in the Actions section is more clear and sufficient.

File: README.md [hunk 5]
```diff
@@ -15,18 +15,48 @@ Just doing `stack install yi` would be akin to unwrapping a box of legos and
...
+  1. editing operations (**BufferM**)
+  2. compositions of buffers (**EditorM**)
+  3. interactions with the OS (**YiM**)
+  
+if and only if the default configurations available do not handle your specific case.
```
Comment (ethercrow) [inline README.md]: "and only if" sounds discouraging if you just wanted to try poking around the configuration example.

File: README.md [hunk 6]
```diff
@@ -15,18 +15,48 @@ Just doing `stack install yi` would be akin to unwrapping a box of legos and
...
+
+#### Advanced configuration examples
```
Comment (ethercrow) [inline README.md]: Let's call these "external". Some of them have only minor changes from the examples.

**Comment by ethercrow** [comment]
I've left some nitpicks. I'd also like a pass from a native speaker, because some parts sound a little off to me, but I'm not sure.

File: README.md (hunk 2 above)
Comment (quickdudley) [inline README.md]: In the section title on line 24: "Configuration" seems more appropriate than "Configuring"

**Comment by ethercrow** [comment]
Thank you!

**Pull request merged**